.buildkite/x-pack/pipeline.xpack.libbeat.yml

```diff
@@ -18,45 +14,176 @@ env:
+          provider: "gcp"
+          image: "${IMAGE_WIN_10}"
+          machineType: "${GCP_WIN_MACHINE_TYPE}"
+          disk_size: 100
+          disk_type: "pd-ssd"
+        artifact_paths:
+          - "x-pack/libbeat/build/*.xml"
+          - "x-pack/libbeat/build/*.json"
+        notify:
+          - github_commit_status:
+              context: "x-pack/libbeat: Windows 10 Unit Tests"
+
+      - label: ":windows: Windows 11 Unit Tests"
+        command: |
+          Set-Location -Path x-pack/libbeat
+          mage -w reader\etw build goUnitTest
+        key: "extended-win-11-unit-tests"
+        agents:
+          provider: "gcp"
+          image: "${IMAGE_WIN_11}"
+          machineType: "${GCP_WIN_MACHINE_TYPE}"
+          disk_size: 100
+          disk_type: "pd-ssd"
+        artifact_paths:
+          - "x-pack/libbeat/build/*.xml"
+          - "x-pack/libbeat/build/*.json"
+        notify:
+          - github_commit_status:
+              context: "x-pack/libbeat: Windows 11 Unit Tests"
+
+      - label: ":windows: Windows 2019 Unit Tests"
+        command: |
+          Set-Location -Path x-pack/libbeat
+          mage -w reader\etw build goUnitTest
+        key: "extended-win-2019-unit-tests"
+        agents:
+          provider: "gcp"
+          image: "${IMAGE_WIN_2019}"
+          machineType: "${GCP_WIN_MACHINE_TYPE}"
+          disk_size: 100
+          disk_type: "pd-ssd"
+        artifact_paths:
+          - "x-pack/libbeat/build/*.xml"
+          - "x-pack/libbeat/build/*.json"
+        notify:
+          - github_commit_status:
+              context: "x-pack/libbeat: Windows 2019 Unit Tests"
 
-  - input: "Input Parameters"
-    key: "input-run-all-stages"
-    fields:
-    - select: "Libbeat - run_xpack_libbeat"
-      key: "run_xpack_libbeat"
-      options:
-        - label: "True"
-          value: "true"
-        - label: "False"
-          value: "false"
-      default: "false"
-    - select: "Libbeat - run_xpack_libbeat_arm_tests"
-      key: "run_xpack_libbeat_arm_tests"
-      options:
-        - label: "True"
-          value: "true"
-        - label: "False"
-          value: "false"
-      default: "false"
-    if: "build.source == 'ui'"
-
-  - wait: ~
-    if: "build.source == 'ui'"
-    allow_dependency_failure: false
-
-  - label: ":linux: Load dynamic x-pack libbeat pipeline"
-    key: "libbeat-pipeline"
-    command: ".buildkite/scripts/generate_xpack_libbeat_pipeline.sh"
-    agents:
-      image: "docker.elastic.co/ci-agent-images/platform-ingest/buildkite-agent-beats-ci:latest"
-    notify:
-      - github_commit_status:
-          context: "${BEATS_PROJECT_NAME}: Load dynamic pipeline's steps"
+  - group: "Extended Tests"
+    key: "extended-tests"
+    if: build.env("BUILDKITE_PULL_REQUEST") == "false" || build.env("GITHUB_PR_LABELS") =~ /.*arm.*/
+    steps:
+      - label: ":linux: Ubuntu ARM64 Unit Tests"
+        key: "extended-arm64-unit-tests"
+        command: |
+          set -euo pipefail
```

Comment:
fyi: this is not needed (`set -euo pipefail`); it will only guard it `cd x-pack/libbeat` fails. The return code of the last command (`mage build unitTest` in this case) will govern success/failure, if `set -euo pipefail` is missing.